### ISSUE TITLE
@vercel/remote-nx is incompatible with nx@16

### ISSUE THREAD

**Issue by TimoBechtel** [vercel/remote-cache#11]
### Which package / area of the repo does the bug impact?

@vercel/remote-nx

### What version are you using?

@vercel/remote-nx@1.0.1

### What package manager are you using / does the bug impact?

pnpm

### What operating system are you using?

Mac

### Describe the Bug

After migrating to nx@16 (using the `nx migrate` command) and trying to run any cachable nx command with @vercel/remote-nx configured, it errors out with `(0 , default_1.default) is not a function`.

Nx has changed their package name scopes to from `@nrwl/*` to `@nx/*`. (https://blog.nrwl.io/nx-16-is-here-69584ec87053#8e44)
The new package is now called `@nx/workspace`. This package (@vercel/remote-nx) still depends on @nrwl/workspace, which causes the error.

As a workaround, I used `pnpm add @nrwl/workspace@npm:@nx/workspace` to alias the old package name to the new.

### Expected Behavior

Should work with nx@16.

### To Reproduce

Configure @vercel/remote-nx with the latest version of nx (16.x). Run any cachable command (e.g. `nx build my-app`).

**Comment by sangalli** [comment]
+1. The lib that is being used by this repo to implement nx cache has been updated recently to support nx 16: https://github.com/NiklasPor/nx-remotecache-custom/commits/v4.0.0

**Comment by pdrmdrs** [comment]
Heya!

Thanks for the issue @TimoBechtel, we also want to update our project to NX@16 but we couldn't because `remote-nx` does not support NX@16 yet.

In a local branch, I updated the `remote-nx` to support NX@16 and also I had to update the `nx-remotecache-custom` to be @4.0.0, as @sangalli mentioned.
Also, I've updated the `examples/nx` as well to use the newer `@nx` instead of `@nrwl`.

How can I contribute and open a PR? I don't think I have permission to do it and the CONTRIBUTING.md file doesn't have any related info about it

I'm happy to open a PR and ask for a review 😃
